src/Symfony/Component/HttpKernel/Bundle/Bundle.php

```diff
@@ -117,6 +118,22 @@ public function getPath()
         return $this->path;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getBundleDir(): string
+    {
+        if (null === $this->dir) {
+            $reflected = new \ReflectionObject($this);
+            $this->dir = \dirname($reflected->getFileName());
+            if (!file_exists($this->dir.'/composer.json') && file_exists(\dirname($this->dir).'/composer.json')) {
```

Comment:
I guess a dynamic detection of the bundle directory through the `composer.json` existence makes more sense today with bundleless apps approach, though you are free to change it at your convenience.

This new method will be necessary to mix both structures until all our depedencies supporting the new convention thus helping bundles migrate smoothly: 
 * we could move first the `public` dir at the bundle root (try out), 
 * next the `templates` dir (try out) 
 * and later the `translations` dir, etc. 
 * and still have e.g. the `Resources/config` dir under `src/` waiting for another packages that don't support the new convention yet.

Reply:
Another approach would be don't implement the method here and force people to implement it on their own. We're already triggering a deprecation about that. That'll feel safe for us. WDYT?

Reply:
Or we remove that new method and keep using `getPath`: ppl that want to put the bundle class inside `src/` would `return \dirname(__DIR__);` here

Reply:
Okay, let's move forward with that option, thanks.

Reply:
Done.